core/src/main/java/com/netflix/iceberg/BaseTableScan.java

```diff
@@ -176,12 +187,15 @@ public InclusiveManifestEvaluator load(Integer specId) {
           matchingManifests,
           manifest -> {
             ManifestReader reader = ManifestReader.read(ops.io().newInputFile(manifest.path()));
+            PartitionSpec spec = ops.current().spec(manifest.partitionSpecId());
             toClose.add(reader);
-            String schemaString = SchemaParser.toJson(reader.spec().schema());
-            String specString = PartitionSpecParser.toJson(reader.spec());
-            ResidualEvaluator residuals = new ResidualEvaluator(reader.spec(), rowFilter);
+            String schemaString = SchemaParser.toJson(spec.schema());
```

Comment:
Not related to this change, but it seems we are converting the PartitionSpec to json and passing it to `BaseFileScanTask` and then, in BaseFileScanTask, reconverting it base to PartitionSpec. Any thoughts on why that is done? Maybe it is worth while to simply pass the PartitionSpec object directly to `BaseFileScanTask` without first converting to json string.  `BaseFileScanTask` does not seem to be called from anywhere else

Reply:
`com.netflix.iceberg.spark.source.Reader.ReadTask` also shows a similar pattern of usage.

Reply:
This is to make the tasks `Serializable` without a huge amount of overhead.